mllib-local/src/main/scala/org/apache/spark/ml/linalg/Matrices.scala

```diff
@@ -455,9 +455,11 @@ class SparseMatrix @Since("2.0.0") (
   require(values.length == rowIndices.length, "The number of row indices and values don't match! " +
     s"values.length: ${values.length}, rowIndices.length: ${rowIndices.length}")
   // The Or statement is for the case when the matrix is transposed
-  require(colPtrs.length == numCols + 1 || colPtrs.length == numRows + 1, "The length of the " +
-    "column indices should be the number of columns + 1. Currently, colPointers.length: " +
-    s"${colPtrs.length}, numCols: $numCols")
+  require(!isTransposed && colPtrs.length == numCols + 1 ||
```

Comment:
This may be clearer if you just have two clauses:

```
if (isTransposed) {
  require(...)
} else {
  require(...)
}
```

That lets you write a shorter specific message in both cases.